homeassistant/components/mqtt/client.py

```diff
@@ -464,16 +464,34 @@ async def _async_unsubscribe(self, topic: str) -> None:
             _raise_on_error(result)
         await self._wait_for_mid(mid)
 
-    async def _async_perform_subscription(self, topic: str, qos: int) -> None:
-        """Perform a paho-mqtt subscription."""
+    async def _async_perform_subscriptions(
+        self, subscriptions: Iterable[tuple[str, int]]
+    ) -> None:
+        """Perform a paho-mqtt subscriptions."""
+
+        def _log_and_subscribe() -> list[tuple[int, int]]:
```

Comment:
Why do we point out that this function logs in its name?

Reply:
The log function was broken with earlier commits, this was restored, but your are right about the naming